### PR TITLE
Add a linter for jpeg turbo linting

### PR DESCRIPTION
**Pull request by hmaarrfk**
xref: https://github.com/conda-forge/conda-forge.github.io/issues/673

Checklist
* [ ] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -931,6 +931,31 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
...
+                    "`jpeg` for improved performance. For more information please see"
+                    "https://github.com/conda-forge/conda-forge.github.io/issues/673"
+                )
+                if msg not in hints:
+                    hints.append(msg)
+
+    hint_jpeg(requirements_section.get("build", []))
+    hint_jpeg(requirements_section.get("host", []))
+    hint_jpeg(requirements_section.get("run", []))
+    for out in outputs_section:
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:951]: `requirements` can be a list instead of a dictionary. Lint in that case too?

Comment (hmaarrfk) [reply conda_smithy/lint_recipe.py:951]: done.

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -931,6 +931,37 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
...
+        if "requirements" not in out:
+            continue
+        requirements = out["requirements"]
+        if isinstance(requirements, list):
+            hint_jpeg(requirements)
+            continue
+        if "build" in requirements:
+            hint_jpeg(requirements["build"])
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:959]:
```suggestion
        hint_jpeg(requirements.get("build", []))
```

Comment (hmaarrfk) [reply conda_smithy/lint_recipe.py]: done. thank you

**Comment by h-vetinari** [comment]
I'm tackling the same issue in more general form in https://github.com/conda-forge/conda-smithy/pull/1734

**Pull request closed without merging**